File: rb/features.rb
# AbdoStore SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module AbdoStoreFeatures
  def self.make_feature(name)
    case name
    when "base"
      AbdoStoreBaseFeature.new
    when "test"
      AbdoStoreTestFeature.new
    else
      AbdoStoreBaseFeature.new
    end
  end
end

File: py/features.py
# AbdoStore SDK feature factory

from feature.base_feature import AbdoStoreBaseFeature
from feature.test_feature import AbdoStoreTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AbdoStoreBaseFeature(),
        "test": lambda: AbdoStoreTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
